### ISSUE TITLE
Possible memory corruption [rt.cpan.org #76921]

### ISSUE THREAD

**Issue by toddr** [cpan-authors/IO-Tty#11]
Migrated from [rt.cpan.org#76921](https://rt.cpan.org/Ticket/Display.html?id=76921) (status was 'new')

Requestors:
* ezra@tsdme.nl

From http://tsdme.nl/portfolio/ on 2012-05-01 10:50:19
:
```
I get the following error message when trying to run Mosh on Cygwin. I
already reported this bug to them, but they told me to take it up with
the IO-Tty team.

"Possible memory corruption: ioctl overflowed 3rd argument at
/usr/lib/perl5/site
_perl/5.10/i686-cygwin/IO/Tty.pm line 57."

The issue at tje mosh project:
https://github.com/keithw/mosh/issues/151

Perl -v: This is perl, v5.10.1 (*) built for i686-cygwin-thread-multi-64int
uname -a: CYGWIN_NT-6.1 cm-pc 1.7.13(0.260/5/3) 2012-04-05 12:43 i686 Cygwin

```

**Comment by tzccinct** [comment]
This issue is reproduced on my Cygwin environment during t/pty_get_winsize.t that has been introduced since IO::Tty 1.14:
http://www.cpantesters.org/cpan/report/55bef3ca-7ba0-1014-89e4-aad475d8a230

Referencing [176154 &ndash; Possible memory corruption: ioctl overflowed 3rd argument](https://bugzilla.redhat.com/show_bug.cgi?id=176154), the following patch has fixed the issue.
```
--- Tty.pm.original     2021-01-22 17:02:35.000000000 +0900
+++ Tty.pm      2021-04-05 00:02:24.487511200 +0900
@@ -71,7 +71,7 @@

 sub get_winsize {
     my $self = shift;
-    ioctl( $self, IO::Tty::Constant::TIOCGWINSZ(), my $winsize = q<> )
+    ioctl( $self, IO::Tty::Constant::TIOCGWINSZ(), my $winsize = chr(0) x $SIZEOF_WINSIZE )
       or croak "Cannot TIOCGWINSZ - $!";
     substr( $winsize, $SIZEOF_WINSIZE ) = "";
     return IO::Tty::unpack_winsize($winsize);
```
`$SIZEOF_WINSIZE` is 8 on my environment. When I tested by `my $winsize = chr(0) x 7`, it failed.

**Comment by tzccinct** [comment]
Regarding the original issue on /usr/lib/perl5/site_perl/5.10/i686-cygwin/IO/Tty.pm line 57, the issue is surely fixed in IO-Tty 1.06:
https://metacpan.org/diff/file?source=RGIERSIG/IO-Tty-1.05&target=RGIERSIG/IO-Tty-1.06

I will make a pull request to do the same approach and coding style as in IO-Tty 1.06.

**Comment by tzccinct** [comment]
I have confirmed this issue is fixed on Cygwin by IO-Tty 1.17.
